### DISCUSSION TITLE
Aim of project?

### DISCUSSION THREAD

**Discussion by conrad82** [danielalvsaaker/tf-viewer#30] [General]
Hi,

I got it working now on Raspberry Pi, and see potential in this tool, but it also seems there is a lot of work remaining

So I was wondering, what is the aim of this project? 
Is it just a showcase to learn some coding, test some tools, or do you have a 1.0 version on the horizon?

I have suggestions for features and improvements, but I don't want to fill up the issues tracker if it is beyond the scope of what you are planning or have the resources to do

**Comment by danielalvsaaker** [comment]
Thanks for asking.

I intend to keep working on this project and wish to reach a version 1.0. The time horizon is hard to estimate, considering it's a student one man project.

I will read every feature request you have, and then consider the work involved. Please elaborate in this thread.

Hopefully the project moves faster when I have more time.

**Reply by conrad82** [reply]
Cool!

Yes, things take time, and for larger / longer projects it is nice to have milestones/goals

At the moment I was thinking of a few things that would be nice

- Archiving of the uploaded FIT files
  - Paperless-ng does this for pdf's for example. It stores all the original files in a separate directory. This is especially nice if something breaks or you have to start over with your self-hosted setup. I did experience a crash when I imported too many FIT files at once, and had to start tf-viewer from scratch. When I tried to reproduce the error by uploading a few FIT files at a time, it did not crash. Maybe a database overload somehow
- Stats for walking
  - I see there are stats for running and cycling, but I use my garmin watch mostly for hiking. It would be nice to have walking included as well.
- Mobile view
  - The view on mobile is a mess. I do think this is a bigger job, but maybe something to include down the line?
- Connect the position in the map and graph
  - It would be nice to see where on the map the graph corresponds to. It seems the map and graph are different entities, so I guess this may not be worth it, at least not for now.

Apart from that I like it so far :) It is nice to be in control of ones own data!

**Reply by danielalvsaaker** [reply]
Thanks for the input. I'm aware of the mobile layout being unusable. I'll return to this project in about a month to work towards a stable version after exams.

**Reply by conrad82** [reply]
Good luck on the exams!

**Comment by danielalvsaaker** [comment]
Version 1.0 is in progress now. The frontend and backend will be completely rewritten, and will by itself solve a lot issues, especially those related to mobile view.

Connecting map and graph, and archiving of files is also planned.
